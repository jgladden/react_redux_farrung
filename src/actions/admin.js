import axios from 'axios';
import * as types from './types';
import { getAdminUrl } from '../config';

export function fetchAdmin(token) {
  const config = {
    headers: {
      'Authorization': `bearer ${token}`
    }
  };
  return dispatch => {
    dispatch(getAdminItems());
    return axios.get(getAdminUrl, config)
      .then(response => dispatch(
        getAdminItemsSuccess(response.data))
      )
      .catch(error => dispatch(
        getAdminItemsError(error.toString()))
      );
  };
}

export const getAdminItems = () => ({
  type: types.GET_ADMIN_ITEMS
});

export const getAdminItemsSuccess = payload => ({
  type: types.GET_ADMIN_ITEMS_SUCCESS,
  payload
});

export const getAdminItemsError = payload => ({
  type: types.GET_ADMIN_ITEMS_ERROR,
  payload
});

export const setDisplayType = payload => ({
  type: types.SET_DISPLAY_TYPE,
  payload
});

export const setDisplayArchived = payload => ({
  type: types.SET_DISPLAY_ARCHIVED,
  payload
});

export const mergeAdminItem = payload => ({
  type: types.MERGE_ADMIN_ITEM,
  payload
});

export const removeAdminItem = payload => ({
  type: types.REMOVE_ADMIN_ITEM,
  payload
});
