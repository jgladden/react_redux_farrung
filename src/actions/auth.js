import axios from 'axios';
import * as types from './types';
import { postAuthUrl } from '../config';

export function submitLogin(authObj) {
  return dispatch => {
    dispatch(postAuth());
    return axios.post(postAuthUrl, authObj)
      .then(response => {
        let error = response.data.error;
        if(!error)
          return dispatch(
            postAuthSuccess(response.data.token)
          );
        return dispatch(
          postAuthError(error)
        );
      })
      .catch(error => dispatch(
        postAuthError(error.toString()))
      );
  };
}

export const postAuth = () => ({
  type: types.POST_AUTH
});

export const postAuthSuccess = payload => ({
  type: types.POST_AUTH_SUCCESS,
  payload
});

export const postAuthError = payload => ({
  type: types.POST_AUTH_ERROR,
  payload
});

export const submitLogout = () => ({
  type: types.AUTH_LOGOUT
});

