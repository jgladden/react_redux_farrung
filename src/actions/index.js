import { jsonFetch } from '../utils/jsonFetch';
import { 
  ADD_PORTFOLIO_TYPE,
  SET_PORTFOLIO_TYPE,
  ADD_PORTFOLIO_ITEM, 
  REMOVE_PORTFOLIO_ITEM 
} from '../constants/action_types';

const api = 'http://farrung.com/api/portfolio1/';

export function fetchPortfolioType(type) {
  let url = api + type;
  return function(dispatch) {
    return jsonFetch(url).then(data => {
      dispatch(addPortfolioType(data));
    });
  };
}

export const addPortfolioType = payload => ({
  type: ADD_PORTFOLIO_TYPE,
  payload
});


export const setPortfolioType = payload => ({
  type: SET_PORTFOLIO_TYPE,
  payload
});

export const addPortfolioItem = payload => ({
  type: ADD_PORTFOLIO_ITEM,
  payload
});

export const removeTodo = payload => ({
  type: REMOVE_PORTFOLIO_ITEM,
  payload
});


