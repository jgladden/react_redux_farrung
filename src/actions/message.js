import axios from 'axios';
import * as types from './types';
import { postMessageUrl } from '../config';

export function submitMessage(messageObj) {
  return dispatch => {
    dispatch(postMessage());
    return axios.post(postMessageUrl, messageObj)
      .then(response => dispatch(
        postMessageSuccess(response.data))
      )
      .catch(error => dispatch(
        postMessageError(error.toString()))
      );
  };
}

export const postMessage = () => ({
  type: types.POST_MESSAGE
});

export const postMessageSuccess = payload => ({
  type: types.POST_MESSAGE_SUCCESS,
  payload
});

export const postMessageError = payload => ({
  type: types.POST_MESSAGE_ERROR,
  payload
});

