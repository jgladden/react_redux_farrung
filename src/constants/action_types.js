export const ADD_PORTFOLIO_ITEM = 'ADD_PORTFOLIO_ITEM';
export const REMOVE_PORTFOLIO_ITEM = 'REMOVE_PORTFOLIO_ITEM';
export const FETCH_PORTFOLIO_TYPE = 'FETCH_PORTFOLIO_TYPE';
export const SET_PORTFOLIO_TYPE = 'SET_PORTFOLIO_TYPE';
export const ADD_PORTFOLIO_TYPE = 'ADD_PORTFOLIO_TYPE';
