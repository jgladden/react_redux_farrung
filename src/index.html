<!DOCTYPE html>
<head>
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name=Keywords content="james gladden, james, gladden, farrung, los angeles web design, los angeles design, web design, design"/>
    <meta name=Description content="Farrung is a Los Angeles based design studio specializing in new media branding, communications and marketing. "/>
    <meta name="viewport" content="width=device-width,initial-scale=1.0,maximum-scale=1.0" />
    <link rel="shortcut icon" type="image/x-icon" href="/favicon.ico">
    <title><%= htmlWebpackPlugin.options.title %></title>
</head>
<body>
    <div id="react-application"></div>
</body>
</html>

