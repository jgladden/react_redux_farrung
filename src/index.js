import './index.scss';
import React from 'react';
import ReactDOM from 'react-dom';
import { Provider } from 'react-redux';
import App from './components/App';
import router from './router';
import { store } from './store';

store.subscribe(router);

ReactDOM.render(
  <Provider store={store}>
    <App />
  </Provider>, 
  document.getElementById('react-application')
);








