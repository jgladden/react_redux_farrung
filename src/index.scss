@import 'sass/normalize';

body {
  font: 14px Arial;
  color: #333;
}

a {
  color: #999;
}

input, textarea {
  color: #666;
}

input {
  display: block;
  border: 1px solid #ccc;
  padding: 5px 10px;
}

.inValidField {
  border: 1px solid #f00;
}

label {
  color: #999;
  font-size: 11px;
  line-height: 11px;
  display: block;
  padding-bottom: 3px;
}
