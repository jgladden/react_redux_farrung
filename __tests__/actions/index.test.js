/*
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import * as actions from '../../src/actions';
import * as types from '../../src/constants/action_types';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);
const initialState = {
  portfolio_type = 'online',
  portfolio_id = '22',
  portfolio: {
    online: {
      22: {
        id: 22,
        title: 'mock title'
      }
    }
  }
};
const store = mockStore(initialState);
*/
