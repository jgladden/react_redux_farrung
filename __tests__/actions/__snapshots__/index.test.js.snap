// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`setPortfolioId dispatches correction action / payload 1`] = `
Array [
  Object {
    "payload": Object {
      "type": "online",
    },
    "type": "SET_PORTFOLIO_TYPE",
  },
  Object {
    "payload": Object {
      "id": "22",
    },
    "type": "SET_PORTFOLIO_ID",
  },
]
`;

exports[`setPortfolioType dispatches correction action / payload 1`] = `
Array [
  Object {
    "payload": Object {
      "type": "online",
    },
    "type": "SET_PORTFOLIO_TYPE",
  },
]
`;
